flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/extraction/ExtractionUtils.java

```diff
@@ -96,7 +96,7 @@ public static boolean isInvokable(Executable executable, Class<?>... classes) {
 		final int paramCount = executable.getParameterCount();
 		final int classCount = classes.length;
 		// check for enough classes for each parameter
-		if (classCount < paramCount || (executable.isVarArgs() && classCount < paramCount - 1)) {
+		if ((!executable.isVarArgs() && classCount < paramCount) || (executable.isVarArgs() && classCount < paramCount - 1)) {
```

Comment:
nit: I would check the first condition to `classCount != paramCount` for an early out